ingress/src/main/java/com/redhat/developer/ingress/producer/KafkaEventPublisher.java

```diff
@@ -0,0 +1,38 @@
+package com.redhat.developer.ingress.producer;
+
+import javax.enterprise.context.ApplicationScoped;
+
+import org.eclipse.microprofile.reactive.messaging.Outgoing;
+import org.jboss.logging.Logger;
+import org.reactivestreams.Publisher;
+
+import com.redhat.developer.infra.utils.CloudEventUtils;
+import com.redhat.developer.ingress.api.exceptions.IngressRuntimeException;
+
+import io.cloudevents.CloudEvent;
+import io.smallrye.mutiny.operators.multi.processors.BroadcastProcessor;
+
+@ApplicationScoped
+public class KafkaEventPublisher {
+
+    private final Logger logger = Logger.getLogger(KafkaEventPublisher.class);
+
+    private final BroadcastProcessor<String> eventSubject = BroadcastProcessor.create();
+
+    public void sendEvent(CloudEvent cloudEvent) {
+        logger.info("Sending event to event queue");
+        String serializedCloudEvent;
+        try {
+            serializedCloudEvent = CloudEventUtils.encode(cloudEvent);
```

Comment:
We can make `CloudEventUtils` throw a subclass of `IngressException` then we don't need to catch and re-throw here.

Reply:
Related to the comments above: `CloudEventUtils` is an utility in the `infra` package that will be used by the `executor` as well so we should not throw `IngressException` or similar in the utility class